src/languageservice/parser/jsonParser07.ts

```diff
@@ -1498,23 +1501,11 @@ function validate(
     node: ASTNode,
     maxOneMatch,
     subValidationResult: ValidationResult,
-    bestMatch: {
-      schema: JSONSchema;
-      validationResult: ValidationResult;
-      matchingSchemas: ISchemaCollector;
-    },
+    bestMatch: IValidationMatch,
     subSchema,
     subMatchingSchemas
-  ): {
-    schema: JSONSchema;
-    validationResult: ValidationResult;
-    matchingSchemas: ISchemaCollector;
-  } {
-    if (
-      !maxOneMatch &&
-      !subValidationResult.hasProblems() &&
-      (!bestMatch.validationResult.hasProblems() || callFromAutoComplete)
```

Comment:
`(!bestMatch.validationResult.hasProblems() || callFromAutoComplete)` didn't solve the previous problem properly
problem was when the order of the subschemas in anyOf changed. 
this previous fix allows 'issues' only on the previous bestMatch